library/pem.c

```diff
@@ -410,7 +410,7 @@ int mbedtls_pem_write_buffer( const char *header, const char *footer,
         return( MBEDTLS_ERR_BASE64_BUFFER_TOO_SMALL );
     }
 
-    if( ( encode_buf = mbedtls_calloc( 1, use_len ) ) == NULL )
+    if( use_len != 0 && ( encode_buf = mbedtls_calloc( 1, use_len ) ) == NULL )
```

Comment:
to make it more readable, please surround the statement `( encode_buf = mbedtls_calloc( 1, use_len ) ) == NULL` with parentheses